src/components/modal/Modal.tsx

```diff
@@ -0,0 +1,40 @@
+import type { JSX } from "react";
+import { ModalActions, ModalContent, ModalSlotContainer, Button, ButtonsContainer } from "./Modal.styles";
+
+interface ModalSlotProps {
+    onClose?: () => void;
+    onConfirm?: () => void;
+    content: JSX.Element;
+    showButtons?: boolean;
+    contentRef?: React.RefObject<any>;
+}
+
+function Modal({ onClose, content, showButtons, onConfirm, contentRef }: ModalSlotProps) {
+    const handleConfirm = () => {
+        // Si hay una ref, intentar llamar al submitForm
+        if (contentRef?.current?.submitForm) {
+            contentRef.current.submitForm();
+        }
+        // También llamar al onConfirm si existe
+        if (onConfirm) {
+            onConfirm();
+        }
+    };
```

Comment:
Aca no hace falta que nosotros le pasemos el ref al modal, porque el modal deberia simplemente recibir cual es la accion a ejecutar cuando se hace click en el boton guardar


```suggestion
    const handleConfirm = () => {
        if (onConfirm) {
            onConfirm();
        }
    };
```